projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -7,219 +7,277 @@ import { useSelect } from '@wordpress/data';
 import { useState, RawHTML, useEffect } from '@wordpress/element';
 import { sprintf, __ } from '@wordpress/i18n';
 
-function formatEditorCanvasContentForCompletion( editor ) {
-	const index = editor.getBlockInsertionPoint().index - 1;
-	const allBlocksBefore = editor.getBlocks().slice( 0, index );
-	if ( ! allBlocksBefore.length ) {
-		return '';
-	}
-	return allBlocksBefore
-		.filter( function ( block ) {
-			return block && block.attributes && block.attributes.content;
-		} )
-		.map( function ( block ) {
-			return block.attributes.content.replaceAll( '<br>', '\n\n' );
-		} )
-		.join( '\n\n' );
-}
+const numberOfCharactersNeeded = 36;
 
-function getSuggestionFromOpenAI(
-	setAttributes,
-	formattedPrompt,
-	setLoadingCompletion,
-	setErrorMessage
-) {
-	const needsAtLeast = 36;
-	if ( formattedPrompt.length < needsAtLeast ) {
-		setLoadingCompletion( false );
-		setErrorMessage(
-			sprintf(
-				/** translators: First placeholder is a number of more characters we need */
-				__(
-					'Please write a longer title or a few more words in the opening preceding the AI block. Our AI model needs %1$d more characters.',
-					'jetpack'
-				),
-				needsAtLeast - formattedPrompt.length
-			)
-		);
-		return;
-	}
-	setErrorMessage( '' );
-
-	const data = { content: formattedPrompt };
-	setLoadingCompletion( true );
-	setAttributes( { requestedPrompt: true } ); // This will prevent double submitting.
-	apiFetch( {
-		path: '/wpcom/v2/jetpack-ai/completions',
-		method: 'POST',
-		data: data,
-	} )
-		.then( res => {
-			setLoadingCompletion( false );
-			const content = res.prompts[ 0 ].text;
+function ShowLittleByLittle( { html, showAnimation } ) {
```

Comment:
Ah, it's easy to miss its a component!
Comment would be nice

Reply:
Nice idea to split it out!

Reply:
It also prevents reloading the whole parent's state at each setTimeout...